08week/newapp/src/components/forms/LoginForm.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Form } from 'semantic-ui-react';
+import RaisedButton from 'material-ui/RaisedButton';
+import TextField from 'material-ui/TextField';
+
+
+class LoginForm extends React.Component {
+    state = {
+        data: {
+          email: '',
+          password: ''
+    },
+        loading: false,
+        errors: {}
+    };
+    //onChange event that is universal for string data
+    //in state change data, use spread to save everything that is in data, then what is changed
+    //which is an e.target
+    onChange = e => this.setState({
+        data: {...this.state.data, [e.target.name]: e.target.value}
```

Comment:
`e.target` doesn't have a `.name` property, so you'll probably be unable to change the input values through typing in them. You could do `e.target.dataset.name` and then also give your TextFields a data name (`<TextField data-name="email">`) to achieve the same effect.